### PR TITLE
Fixing race condition wherein glove calibration tries to update finge…

### PR DESCRIPTION
**Pull request by ethanfowler**
…rtip mapping before it is ready

## Proposed changes

Please explain the changes you made. Add pictures or videos to explain them better if appropriate.

## Checklist

Before posting a PR ensure that from each of the below categories **AT LEAST ONE BOX HAS BEEN CHECKED**. If more than one category is applicable then more can be checked. Also ensure that the proposed changes have been filled out with relevant information for reviewers.

## Tests

- [ ] No tests required to be added. (For small changes that will be tested by CI/CD infrastructure).
- [ ] Added/Modified automated and PhantomHand CI tests (if a new class is added (Python or C++), the interface of that class must be unit tested).
- [ ] Manually tested in simulation (if simulation specific or no hardware required to test the functionality). 
- [x] Manually tested on hardware (if hardware specific or related).

## Documentation

- [ ] No documentation required to be added.
- [x] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc).
- [ ] Updated documentation (For changes to pre-existing features mentioned in the documentation).

### CODE REVIEW

File: src/polhemus_ros_driver/knuckle_position_calibration_action_server.py [hunk 1]
```diff
@@ -255,29 +255,50 @@ def _publish_calibration(self, hands: List[Hand], save: bool = True):
...
+            fingers_used_for_thumbscaling = [finger for finger in fingers_with_length if finger != "lf"]
+            # If there are fingers with lengths
+            if fingers_with_length:
+                # Enable scaling
+                new_fingertip_teleop_config = {"scaling": True}
+                # Calculate the scaling factor for each finger
+                for finger in fingers_with_length:
+                    if hand.finger_data[finger]['length']:
+                        finger_scaling = 0.096 / (hand.finger_data[finger]['length'][-1] + 0.01)
+                        new_fingertip_teleop_config[finger + '_scaling_factor'] = finger_scaling
```
Comment (jeferal) [inline src/polhemus_ros_driver/knuckle_position_calibration_action_server.py]: Should we use f-string here instead?

File: src/polhemus_ros_driver/knuckle_position_calibration_action_server.py [hunk 2]
```diff
@@ -255,29 +255,50 @@ def _publish_calibration(self, hands: List[Hand], save: bool = True):
...
+            # If there are fingers with lengths that are not the little finger
+            if fingers_used_for_thumbscaling:
+                # Calculate the average scaling factor, to be used for the thumb
+                new_fingertip_teleop_config['th_scaling_factor'] = (sum([
+                    new_fingertip_teleop_config[finger + '_scaling_factor'] for finger in
```
Comment (jeferal) [inline src/polhemus_ros_driver/knuckle_position_calibration_action_server.py]: and here

**Review by bdenoun — APPROVED** [review]
Approved after Jesus' comment are accounted for

File: src/polhemus_ros_driver/knuckle_position_calibration_action_server.py (hunk 1 above)
Comment (ethanfowler) [reply src/polhemus_ros_driver/knuckle_position_calibration_action_server.py]: Addressed in https://github.com/shadow-robot/polhemus_ros_driver/pull/50/commits/005a21b3d8b2dd00a868048974104c171209f2d7

File: src/polhemus_ros_driver/knuckle_position_calibration_action_server.py (hunk 2 above)
Comment (ethanfowler) [reply src/polhemus_ros_driver/knuckle_position_calibration_action_server.py]: Addressed in https://github.com/shadow-robot/polhemus_ros_driver/pull/50/commits/005a21b3d8b2dd00a868048974104c171209f2d7

**Review by jeferal — APPROVED** [review]

**Pull request merged**